flink-connectors/flink-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/HBaseSink.java

```diff
@@ -0,0 +1,554 @@
+	 * <p>A use case for this is in migration between Flink versions or changing the jobs in a way that changes the
+	 * automatically generated hashes. In this case, providing the previous hashes directly through this method (e.g.
+	 * obtained from old logs) can help to reestablish a lost mapping from states to their target operator.
+	 *
+	 * @param uidHash The user provided hash for this operator. This will become the JobVertexID, which is shown in the
+	 *                 logs and web ui.
+	 * @return The operator with the user provided hash.
+	 */
+	@PublicEvolving
+	public HBaseSink<IN> setUidHash(String uidHash) {
+		getSinkTransformation().setUidHash(uidHash);
+		return this;
+	}
+
+	/**
+	 * Sets the parallelism for this sink. The degree must be higher than zero.
+	 *
+	 * @param parallelism The parallelism for this sink.
+	 * @return The sink with set parallelism.
+	 */
+	public HBaseSink<IN> setParallelism(int parallelism) {
+		getSinkTransformation().setParallelism(parallelism);
+		return this;
+	}
+
+	/**
+	 * Turns off chaining for this operator so thread co-location will not be
+	 * used as an optimization.
+	 * <p/>
+	 * <p/>
+	 * Chaining can be turned off for the whole
+	 * job by {@link org.apache.flink.streaming.api.environment.StreamExecutionEnvironment#disableOperatorChaining()}
+	 * however it is not advised for performance considerations.
+	 *
+	 * @return The sink with chaining disabled
+	 */
+	public HBaseSink<IN> disableChaining() {
+		getSinkTransformation().setChainingStrategy(ChainingStrategy.NEVER);
+		return this;
+	}
+
+	/**
+	 * Sets the slot sharing group of this operation. Parallel instances of
+	 * operations that are in the same slot sharing group will be co-located in the same
+	 * TaskManager slot, if possible.
+	 *
+	 * <p>Operations inherit the slot sharing group of input operations if all input operations
+	 * are in the same slot sharing group and no slot sharing group was explicitly specified.
+	 *
+	 * <p>Initially an operation is in the default slot sharing group. An operation can be put into
+	 * the default group explicitly by setting the slot sharing group to {@code "default"}.
+	 *
+	 * @param slotSharingGroup The slot sharing group name.
+	 */
+	public HBaseSink<IN> slotSharingGroup(String slotSharingGroup) {
+		getSinkTransformation().setSlotSharingGroup(slotSharingGroup);
+		return this;
+	}
+
+	/**
+	 * Writes a DataStream into a HBase database.
+	 *
+	 * @param input input DataStream
+	 * @param <IN>  input type
+	 * @return HBaseSinkBuilder, to further configure the sink
+	 */
+	public static <IN> HBaseSinkBuilder<IN> addSink(org.apache.flink.streaming.api.scala.DataStream<IN> input) {
+		return addSink(input.javaStream());
+	}
+
+	/**
+	 * Writes a DataStream into a HBase database.
+	 *
+	 * @param input input DataStream
+	 * @param <IN>  input type
+	 * @return HBaseSinkBuilder, to further configure the sink
+	 */
+	public static <IN> HBaseSinkBuilder<IN> addSink(DataStream<IN> input) {
```

Comment:
I don't think we need this class, it has the same methods with `DataStream` excepts this utility method. However, if we only introduce Row type sink, we won't need this method. IMO, a builder to create a hbase SinkFunction is enough for the first version.

Reply:
Sounds good.